Plugins/explicitDependencyImportCheckPlugin/ExplicitDependencyImportCheckPlugin.swift

```diff
@@ -4,18 +4,30 @@ import PackagePlugin
 @main
 struct ExplicitDependencyImportCheckPlugin: BuildToolPlugin {
     func createBuildCommands(context: PluginContext, target: any Target) async throws -> [Command] {
-        let dependeniesNames = findSpmTargetDependencies(fromTarget: target)
-        let sitrepResponse = try findImportsUsedInCode(context: context, path: target.directory.string)
-        let transitiveDependencies = sitrepResponse.subtracting(dependeniesNames)
-        let transitiveDependenciesWithoutCommon = transitiveDependencies.subtracting(systemLibraries)
-        guard transitiveDependenciesWithoutCommon.isEmpty == false else { return [] }
-        return [
-            try exportTransitiveDependenciesAsErrors(
-                transitiveDeps: transitiveDependenciesWithoutCommon,
-                forTarget: target,
-                inContext: context
-            )
-        ]
+        let targetDependencies = findSpmTargetDependencies(fromTarget: target)
+        let importsInCode = try findImportsUsedInCode(context: context, path: target.directory.string)
+        let transitiveDependencies = findTransitiveDependencies(imports: importsInCode, targetDependencies: targetDependencies)
+        let unusedTargetDependencies = findUnusedTargetDependencies(imports: importsInCode, targetDependencies: targetDependencies)
+
+        return (try exportTransitiveDependenciesAsErrors(
+                   transitiveDeps: transitiveDependencies,
+                   forTarget: target,
+                   inContext: context
+                )) +
+                (try exportUnusedDependenciesAsWarnings(
```

Comment:
Great job 💯 
I think it would be also great to update the readme to include this feature addition

Reply:
Ah yeah. Will do.